modin/core/storage_formats/pandas/aggregations.py

```diff
@@ -0,0 +1,588 @@
+            #  parts_without_nans.repeat(parts_with_nans.shape[0]):
+            #    sum  pow2_sum  count
+            # a  1    3         5
+            # b  1    3         5
+            # a  2    4         6
+            # b  2    4         6
+            #
+            # And then reshape it using the "Fortran" order:
+            #  parts_without_nans.reshape(parts_with_nans.shape, order="F"):
+            #    sum   pow2_sum  count
+            #    a  b  a  b      a  b
+            # a  1  2  3  4      5  6
+            # b  1  2  3  4      5  6
+            # After that the shapes & orders are aligned and we can perform the summation
+            .repeat(repeats=len(parts_with_nans), axis=0).reshape(
+                parts_with_nans.shape, order="F"
+            )
+        )
+        replace_values = parts_with_nans + parts_without_nans
+
+        if not total_agg.values.flags.writeable:
+            # making a copy if the buffer is read-only as
+            # we will need to modify `total_agg` inplace
+            total_agg = total_agg.copy()
+        total_agg.values[:, na_agg_idxs] = replace_values
+
+        return total_agg
+
+    @staticmethod
+    def _build_corr_table_nan(
+        sum_of_pairwise_mul: pandas.DataFrame,
+        means: pandas.DataFrame,
+        sums: pandas.DataFrame,
+        count: pandas.DataFrame,
+        std: pandas.DataFrame,
+        cols: pandas.Index,
+        min_periods: int,
+    ) -> pandas.DataFrame:
+        """
+        Build correlation matrix for a DataFrame that had NaN values in it.
+
+        Parameters
+        ----------
+        sum_of_pairwise_mul : pandas.DataFrame
+        means : pandas.DataFrame
+        sums : pandas.DataFrame
+        count : pandas.DataFrame
+        std : pandas.DataFrame
+        cols : pandas.Index
+        min_periods : int
+
+        Returns
+        -------
+        pandas.DataFrame
+            Correlation matrix.
+        """
+        res = pandas.DataFrame(index=cols, columns=cols, dtype="float")
+
+        for col_idx, i in enumerate(cols):
+            mul_row = sum_of_pairwise_mul.loc[i]
+            mean_row = means.loc[i]
+            sum_row = sums.loc[i]
+            count_row = count.loc[i]
+            std_row = std.loc[i]
+
+            if count_row[i] >= min_periods:
+                res.loc[i, i] = 1.0
+
+            for j in cols[col_idx + 1 :]:
+                if count_row.loc[j] < min_periods:
+                    # Leave the value to be NaN
+                    continue
+                top = (
+                    mul_row.loc[j]
+                    - means.loc[j, i] * sum_row.loc[j]
+                    - mean_row.loc[j] * sums.loc[j, i]
+                    + count_row.loc[j] * mean_row.loc[j] * means.loc[j, i]
+                )
+                down = std_row.loc[j] * std.loc[j, i]
+                corr_ij = top / down
+                res.loc[i, j] = corr_ij
+                res.loc[j, i] = corr_ij
```

Comment:
If the correlation coefficients are symmetric, then why is `sums` matrix not symmetric?

Reply:
Indeed, the `sums`, `means`, `std` matrices are not symmetric, however, the `sum_of_pairwise_mul` and `count` matrices are always symmetric. From these facts, we can infer that the resulted correlation matrix is also always symmetrical:

Consider this formula for the correlation coefficient:
```
top[a, b] = mul[a, b] - mean[b, a] * sum[a, b] - mean[a, b] * sum[b, a] + count[a, b] * mean[a, b] * mean[b, a]
down[a, b] = std[a, b] * std[b, a]
corr[a, b] = top[a, b] / down[a, b]
```
As it's seen, the `down[a, b] == down[b, a]`. As we know that `mul` and `count` are symmetric matrices, let's replace their components with simply `M` and `C`, then we will have the following expression for the `top[a, b]`:
```
top[a, b] = M - mean[b, a] * sum[a, b] - mean[a, b] * sum[b, a] + C * mean[a, b] * mean[b, a]
```
As you can now see, the `top[a, b] == top[b, a]` as well, meaning that the correlation matrix itself should also be symmetric.

Reply:
yes, I see, thanks!